R/nnmf_sparse.R

```diff
@@ -0,0 +1,267 @@
+#' Non-Negative Matrix Factorization Signal Extraction with lasso Penalization
+#'
+#' `step_nnmf_sparse()` creates a *specification* of a recipe step
+#'  that will convert numeric data into one or more non-negative
+#'  components.
+#'
+#' @inheritParams step_pca
+#' @inheritParams step_center
+#' @param num_comp The number of components to retain as new predictors. If
+#'  `num_comp` is greater than the number of columns or the number of possible
+#'  components, a smaller value will be used.
+#' @param penalty A non-negative number used as a penalization factor for the
+#' loadings. Values are usually between zero and one.
+#' @param options A list of options to `nmf()` in the RcppML package. That
+#'  package has a separate function `setRcppMLthreads()` that controls the
+#'  amount of internal parallelization. **Note** that the argument `A`, `k`,
+#'  `L1`, and `seed` should not be passed here.
+#' @param res A matrix of loadings is stored here, along with the names of the
+#'  original predictors, once this preprocessing step has been trained by
+#'  [prep.recipe()].
```

Comment:
```suggestion
#'  [`prep()`][prep.recipe()].
```
We should try to make links like this all the time, because people find the method names as links, instead of the function names, confusing.

Reply:
I'll make another PR related to this since we have stuff like ``` [`tidy()`] ``` in various places